File: Python/addTwoNumbers.py

l1 = [9,9,9,9,9,9,9] 
l2 = [9,9,9,9]

def addTwoNumbers(l1,l2):
    # def listNode_to_list(node):
    #         result = []
    #         while node:
    #             result.append(node.val)
    #             node = node.next
    #         return result
    # def list_to_listNode(lst):
    #         dummy = ListNode()
    #         current = dummy
    #         for value in lst:
    #             current.next = ListNode(value)
    #             current = current.next
    #         return dummy.next

    #     array_l1 = listNode_to_list(l1)
    #     array_l2 = listNode_to_list(l2)
    num1 = ""
    num2 = ""
    for x in l1[::-1]:
        num1 += str(x)
    for x in l2[::-1]:
        num2 += str(x)
    sum = str(int(num1) + int(num2))
    arr = []
    for x in sum:
        arr.append(int(x))

    return arr[::-1]

print(addTwoNumbers(l1,l2))    

File: JavaScript/addTwoNumbers.js

  function ListNode(val, next) {
     this.val = (val===undefined ? 0 : val)
     this.next = (next===undefined ? null : next)
  }


 let l1 = [2,4,3]
 let l2 = [5,6,4]


function addTwoNumbers(l1, l2) {
    function listNodeToList(node) {
        let result = [];
        while (node) {
            result.push(node.val);
            node = node.next;
        }
        return result;
    }
    
    function listToListNode(lst) {
        let dummy = new ListNode();
        let current = dummy;
        for (let value of lst) {
            current.next = new ListNode(value);
            current = current.next;
        }
        return dummy.next;
    }
    let arrayL1 = listNodeToList(l1);
    let arrayL2 = listNodeToList(l2);

    let num1 = arrayL1.reverse().join("");
    let num2 = arrayL2.reverse().join("");

    let sum = (BigInt(num1) + BigInt(num2)).toString();
    let arr = Array.from(sum, Number).reverse();

    return listToListNode(arr);
}

console.log(addTwoNumbers(l1, l2))